### DISCUSSION TITLE
About Legacy BIOS and UEFI

### DISCUSSION THREAD

**Discussion by saruru8441** [ltsp/ltsp#697] [Q&A] [answered]
Hi
I am using the previous version of LTSP (Ubuntu 18.04) and the new LTSP (Ubuntu 20.04).
The previous version of the LTSP client boots successfully in legacy BIOS with PXE.
The new LTSP client is UEFI. It is also working fine, but we are consolidating to one new LTSP because we think it is inefficient to use it on two servers, but the legacy BIOS client will not boot up.
The only place it won't boot is
PXE NETWORK BOOT xxxxxxxx
Client MAC ADD : xx xx xx xx xx GUID: xxxx-xxxxxx-xxxx-xxxx-xxxxxxxxx
Client IP :192.168.67.37 MASK: 255.255.255.0 DHCP IP: 192.168.67.1
PXE->EB: !PXE at 9D90:0040,entry point at 9D90:00CC
                UNDI code segment 9D90:2300, data segment 8EBB:ED50(570-639kB)
　　　　UNDI demice is PCI 00:0A.0, type DIX+802.3
               570kB free base memory after PXE unload 
iPXE initialising devices.....
Stopped at the above.

UEFI clients can boot without problems on the same server.

Is there any way to get the new LTSP (Ubuntu 20.04) to work with the client that was running on the previous version?

I know this may be very simple for you all, but can you help?

I'm using translation software to translate, so if it's hard to understand, my apologies.
Thank you very much.

**Comment by alkisg** [comment]
Hi, try this:

https://github.com/ltsp/ltsp/discussions/276#discussioncomment-102218

**Reply by saruru8441** [reply]
Thank you very much, Mr. alkisg.

We have not been able to test it on an actual machine, but we have done so with HYPER-V.
I have made some progress, but it still does not start. Is there anything I should change?
UNDI code segment 9D90:2300, data segment 8EBB:ED50(570-639kB)
　　　　UNDI demice is PCI 00:0A.0, type DIX+802.3
570kB free base memory after PXE unload
iPXE initialising devices.....WARNING:Using Legacy NIC wrapper on xx:xx:xx:xx:xx(MAC ADD)

Thanks again for your prompt replies.

**Comment by alkisg** [comment, marked as answer]
If you're testing with hyper-v, it might just have a bad UEFI implementation.
Try changing the NIC type, and/or try to boot with https://boot.ipxse.org/ipxe.iso.
You can also ask in the #ipxe libera IRC channel for ipxe issues with hyper-v.

**Reply by saruru8441** [reply]
Thank you very much,
We will try to adjust the components so that we can test it on the actual machine.

I appreciate your very fast reply.

**Reply by saruru8441** [reply]
Hello, alkis.
We have verified this on the actual machine.
The problem has been solved with the information you provided.

Thank you for all your help.
You are the best!
